lib/generic_server_operation.rb

```diff
@@ -0,0 +1,41 @@
+module ManageIQ::Providers::Hawkular
+  class GenericServerOperation
```

Comment:
Now that you have this one here, I think that we should agree on what to place in MiQ vs in hawkular-client. For me, this one is candidate to be placed in hawkular-client., making it available to any user of the gem without reinventing the wheel.